### PR TITLE
Added support for route/action in array form

### PR DESCRIPTION
**Pull request by DODMax**
This allows to use nested resources by adding support of routes/action in array form as mentioned in issue https://github.com/dwightwatson/bootstrap-form/issues/43

Example:

```
BootForm::open([
    'model' => $application,
    'store' => array('ApplicationController@store', $project),
    'update' => array('ApplicationController@update', $project)
])
```

### CODE REVIEW

File: src/BootstrapForm.php [hunk 1]
```diff
@@ -143,13 +143,19 @@ protected function model($options)
         // If the form is passed a model, we'll use the update route to update
         // the model using the PUT method.
         if (!is_null($options['model']) && $options['model']->exists) {
-            $route = Str::contains($options['update'], '@') ? 'action' : 'route';
-
-            $options[$route] = [$options['update'], $options['model']->getRouteKey()];
+            $update = is_array($options['update']) ? $options['update'][0] : $options['update'];
+            $route = Str::contains($update, '@') ? 'action' : 'route';
+            
+            if (is_array($options['update'])) {
```
Comment (dwightwatson) [inline src/BootstrapForm.php]: Would it work if you just cast `$options['update']` to an array and then merge it? Would save you needing the if/else statement.

`$options[$route] = array_merge((array) $options['update'], [$options['model']->getRouteKey()]);`

**Comment by dwightwatson** [comment]
Thanks for taking the time to do this.

No chance you could add tests for this is there?

**Comment by DODMax** [comment]
It works like a charm, sorry for being so slow to test

**Comment by vis-bp** [comment]
For the nested resources, isn't there is a mistake in commit https://github.com/dwightwatson/bootstrap-form/pull/46/commits/467eb885253557cfb873736a33639004b0ad10c8 in line 146. 

It should probably read
`$update = is_array($options['update']) ? $options['update'][0] : $options['update'];`

**Comment by vis-bp** [comment]
The changes in `protected function model($options)`seem to add the intended functionality - it would be nice if this codes makes it to the master...

**Comment by dwightwatson** [comment]
My concern is that there are no tests that cover the original and new functionality to ensure nothing is broken for existing users. I haven't had a good chance to pull this down and have a crack at those tests myself.

In addition, it looks like new commits have been added that include support for help text that I do not intend to merge. (Bootstrap already has it's own help text class for forms and popovers, and I do not think  it is appropriate for this package to ship with CSS files).

**Comment by DODMax** [comment]
Hello,

I will branch the fix out and make another pull request.
I will also try to look into the test cases.

**Comment by Benestar** [comment]
Is there any chance this PR gets merged?

**Comment by Benestar** [comment]
Wow thanks a lot for the fast response :+1:

**Pull request merged**